src/main/java/picard/sam/markduplicates/MarkDuplicates.java

```diff
@@ -772,6 +774,8 @@ private void generateDuplicateIndexes(final boolean useBarcodes, final boolean i
             if (TAG_DUPLICATE_SET_MEMBERS) {
                 addRepresentativeReadIndex(nextChunk);
             }
+        }  else if (nextChunk.size() == 1) {
```

Comment:
Now that the common code shared between lines 761-766 and lies 772-779 is longer it seems worthwhile making a separate method to avoid code duplication. The IDE can do this for you, here's one way to do it:
```
        for (final ReadEndsForMarkDuplicates next : this.pairSort) {
            if (firstOfNextChunk != null && areComparableForDuplicates(firstOfNextChunk, next, true, useBarcodes)) {
                nextChunk.add(next);
            } else {
                handleChunk(nextChunk);
                nextChunk.clear();
                nextChunk.add(next);
                firstOfNextChunk = next;
            }
        }
        handleChunk(nextChunk);
```

```
    private void handleChunk(List<ReadEndsForMarkDuplicates> nextChunk) {
        if (nextChunk.size() > 1) {
            markDuplicatePairs(nextChunk);
            if (TAG_DUPLICATE_SET_MEMBERS) {
                addRepresentativeReadIndex(nextChunk);
            }
        } else if (nextChunk.size() == 1) {
            addSingletonToCount(libraryIdGenerator);
        }
    }
```